vault/core.go

```diff
@@ -648,6 +648,15 @@ type Core struct {
 	effectiveSDKVersion string
 
 	rollbackPeriod time.Duration
+
+	pendingRemovalMountsAllowed bool
+
+	// majorUpgradeInProgress tells Vault how to handle mount entries
+	// associated with deprecated builtins. If true, this results in a core
+	// shutdown on unseal.  Otherwise, it informs setupCredentials and
+	// setupMounts to continue processing the mount entry, but skips backend
+	// initialization.
+	majorUpgradeInProgress bool
```

Comment:
This name seems very misleading to me.  Outside of the contest of this PR, I would think based on the name that it relates to vault upgrade operations, e.g. the replacement of nodes, or internal data migrations relating to upgrades.  How about something more specific, like failUnsealIfDeprecatedBuiltinsExist?